### ISSUE TITLE
Burning tables makes wholes

### ISSUE THREAD

**Issue by BROtania** [cataclysmbnteam/Cataclysm-BN#3744]
### Describe the bug

if you set fire to furniture (I've only tested f_table, f_coffee_table, f_desk), then when burning, some of the floors under them collapse.gorenje If you use this, then instead of a bunch of resources and many hours of digging, you can just set fire to furniture. works only with some types of floors.

### Steps To Reproduce

1. create any floor
2. place table on this floor
3. burn table
4. wait ~1 hour
5. now you have a whole

### Screenshots

![Снимок экрана (27)](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/73956193/e2b667b9-12c1-48c3-89e7-b336700c926a)
![Снимок экрана (29)](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/73956193/912b0d6d-c49c-4163-bbda-6540c32fd789)

### Versions and configuration

OS: Windows 10 (win_64x).
Versions: all versions from last stable (Cataclysm-BN 0.4), or maybe it work on older versions

**Comment by scarf005** [comment]
@RoyalFox2140 could this be solved soley using JSON?

**Comment by RoyalFox2140** [comment]
Nope. I've set paper walls on fire last night and got this bug. Look at the ter_set field here. Something bad is going on because it doesn't respect what furniture or terrain SHOULD become on destruction.

```json
{
    "type": "terrain",
    "id": "t_paper",
    "name": "paper wall",
    "description": "A huge wall of pulpy mass, covered in sticky wasp saliva.  You could smash it down effortlessly.",
    "symbol": "#",
    "color": "white",
    "move_cost": 0,
    "coverage": 100,
    "flags": [ "FLAMMABLE_ASH", "NOITEM", "WALL", "NO_SCENT" ],
    "bash": {
      "str_min": 1,
      "str_max": 6,
      "sound": "rrrrip!",
      "sound_fail": "slap!",
      "sound_vol": 8,
      "sound_fail_vol": 4,
      "ter_set": "t_floor_paper"
    }
  },
  ```
